File: 11. Container With Most Water/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let maxArea = 0
    let leftPtr = 0
    let rightPtr = height.length -1
    
    while(leftPtr<rightPtr){
       let tempMaxArea= (Math.min(height[leftPtr],height[rightPtr])) *(rightPtr-leftPtr)
        if(tempMaxArea>maxArea){
            maxArea = tempMaxArea
        }
        
        if(height[leftPtr]<height[rightPtr]){
            leftPtr+=1
        }else{
            rightPtr-=1
        }
    }
    
    return maxArea
    
};

File: 11. Container With Most Water/11. Container With Most Water/11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_area = 0
        left_ptr = 0
        right_ptr = len(height) -1
        
        while(left_ptr < right_ptr):
            temp_max_area = (min(height[left_ptr], height[right_ptr])) * (right_ptr - left_ptr)
            if(temp_max_area > max_area):
                max_area = temp_max_area
            
            if(height[left_ptr] < height[right_ptr]):
                left_ptr += 1
            else:
                right_ptr -= 1
        
        return max_area
